backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_BOM.java

```diff
@@ -109,4 +121,69 @@ public void validateBOMAttributes(final I_PP_Product_BOM productBom)
 		productPlanningDAO.retrieveProductPlanningForBomVersions(productBOMVersionsId)
 				.forEach(productPlanning -> productBOMService.verifyBOMAssignment(productPlanning, productBom));
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_PP_Product_BOM.COLUMNNAME_ValidFrom, I_PP_Product_BOM.COLUMNNAME_ValidTo })
+	public void preventBOMVersionsOverlapping(final I_PP_Product_BOM productBom)
+	{
+		if (productBom.getValidTo() != null && productBom.getValidTo().before(productBom.getValidFrom()))
+		{
+			throw new AdempiereException(MSG_VALID_TO_BEFORE_VALID_FROM);
+		}
+
+		final ProductBOMVersionsId bomVersionsId = ProductBOMVersionsId.ofRepoId(productBom.getPP_Product_BOMVersions_ID());
+		final BOMType bomType = BOMType.ofNullableCode(productBom.getBOMType());
+		final Optional<I_PP_Product_BOM> latestBOMVersions = bomDAO.getLatestBOMRecordByVersionAndType(bomVersionsId, ImmutableSet.of(bomType));
```

Comment:
why not retrieve and check all BOMs of bomVersionsId and bomType?
why check only the latest one?

Reply:
code updated